__tests__/backgroundScript.test.js

```diff
@@ -95,8 +98,62 @@ describe("backgroundScript", () => {
   });
 
   test("Should log an error if the selector was not found", () => {
-    const consoleSpy = vi.spyOn(console, 'log').mockReturnValue();
-    setupDOM('<html></html>', ['h1']);
+    const consoleSpy = vi.spyOn(console, "log").mockReturnValue();
+    setupDOM("<html></html>", ["h1"]);
     expect(consoleSpy).toHaveBeenCalledOnce();
   });
+
+  describe("amountFound", () => {
+    const html = `
+    <div id="container" class='red primary'>
+      <div>
+        <span class='blue primary'> Span </span>
+        <div>
+          <h4 class='primary'> H4 </h4>
+        </div>
+      </div>
+      <h2 id="subheader" class='red'>This is not a h1</h2>
+    </div>
+  `;
+
+    test("should be equal to 0 without selectors", () => {
+      const { amountFound } = setupDOM(null, []);
+      expect(amountFound).toBe(0);
+    });
+
+    test("should be equal to 1 when one h1 is present", () => {
+      const { amountFound } = setupDOM(null, ["h1"]);
+      expect(amountFound).toBe(1);
+    });
+
+    test("should be equal to 1 when one h2 is present", () => {
+      const { amountFound } = setupDOM(null, ["h2"]);
+      expect(amountFound).toBe(1);
+    });
```

Comment:
Whats the difference between those two tests?
Please review all tests again and make sure that we are not testing internal functions, but our implementation.

Reply:
Different selectors, the html was set up with default value above

Reply:
Yeah but its still a selector, does not matter if its span or h2, its the same object from js/dom perspective

Reply:
Maybe Ill try explain that with test context:

1. Given you have HTML context that contains X element.
2. When you setupDOM and run the function
3. Then you expect amount to be found N times of X element

Its the same test in both cases, you just use different element, it does not include any new value.

Reply:
okey, I will remove one of them